stored_requests/backends/file_fetcher/fetcher.go

```diff
@@ -16,45 +16,93 @@ import (
 // This expects each file in the directory to be named "{config_id}.json".
 // For example, when asked to fetch the request with ID == "23", it will return the data from "directory/23.json".
 func NewFileFetcher(directory string) (stored_requests.Fetcher, error) {
-	storedReqData, err := collectStoredData(directory + "/stored_requests")
-	if err != nil {
-		return nil, err
-	}
-	storedImpData, err := collectStoredData(directory + "/stored_imps")
-	if err != nil {
-		return nil, err
-	}
-
-	return &eagerFetcher{storedReqData, storedImpData}, nil
+	storedData, err := collectStoredData(directory, FileSystem{make(map[string]FileSystem), make(map[string]json.RawMessage)}, nil)
+	return &eagerFetcher{storedData}, err
 }
 
 type eagerFetcher struct {
-	storedReqs map[string]json.RawMessage
-	storedImps map[string]json.RawMessage
+	FileSystem FileSystem
 }
 
 func (fetcher *eagerFetcher) FetchRequests(ctx context.Context, requestIDs []string, impIDs []string) (map[string]json.RawMessage, map[string]json.RawMessage, []error) {
-	errs := appendErrors("Request", requestIDs, fetcher.storedReqs, nil)
-	errs = appendErrors("Imp", impIDs, fetcher.storedImps, errs)
-	return fetcher.storedReqs, fetcher.storedImps, errs
+	storedRequests := fetcher.FileSystem.Directories["stored_requests"].Files
+	storedImpressions := fetcher.FileSystem.Directories["stored_imps"].Files
+	errs := appendErrors("Request", requestIDs, storedRequests, nil)
+	errs = appendErrors("Imp", impIDs, storedImpressions, errs)
+	return storedRequests, storedImpressions, errs
 }
 
-func collectStoredData(directory string) (map[string]json.RawMessage, error) {
+func (fetcher *eagerFetcher) FetchCategories(primaryAdServer, publisherId, iabCategory string) (string, error) {
+	for k, v := range fetcher.FileSystem.Directories {
```

Comment:
It's good that this is looking up the category now, however since FileSystem uses maps you can just index the map by primaryAdServer and then fileName instead of iterating over all entries in the maps to look for a matching key. Indexing will return a second value that is boolean of whether or not key exists, so you can write `if adServerMappings, found := fetcher.FileSystem.Directories[primaryAdServer]; found {` instead of `for k,v := range fetcher.FileSystem.Diretories { if k == primaryAdServer {`

Reply:
You are right, thats all maps now, I removed second loop too. Good catch!